uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/FileNavigatorServiceImpl.java

```diff
@@ -60,24 +60,25 @@ public NavigatorContent listContent(final org.uberfire.backend.vfs.Path _path) {
         final ArrayList<org.uberfire.backend.vfs.Path> breadcrumbs = new ArrayList<>();
 
         Path path = Paths.convert(_path);
-        final DirectoryStream<Path> stream = ioService.newDirectoryStream(path);
-
-        for (final Path activePath : stream) {
-            final VersionAttributeView versionAttributeView = ioService.getFileAttributeView(activePath,
-                                                                                             VersionAttributeView.class);
-            int index = versionAttributeView.readAttributes().history().records().size() - 1;
-
-            final String authorEmail = versionAttributeView.readAttributes().history().records().get(index).email();
-            final String author = versionAttributeView.readAttributes().history().records().get(index).author();
-            final String comment = versionAttributeView.readAttributes().history().records().get(index).comment();
-
-            final String time = p.format(new Date(Files.getLastModifiedTime(activePath).toMillis()));
-            result.add(new DataContent(Files.isDirectory(activePath),
-                                       comment,
-                                       author,
-                                       authorEmail,
-                                       time,
-                                       Paths.convert(activePath)));
+        try (final DirectoryStream<Path> stream = ioService.newDirectoryStream(path)) {
+
+            for (final Path activePath : stream) {
+                final VersionAttributeView versionAttributeView = ioService.getFileAttributeView(activePath,
+                                                                                                 VersionAttributeView.class);
+                int index = versionAttributeView.readAttributes().history().records().size() - 1;
+
+                final String authorEmail = versionAttributeView.readAttributes().history().records().get(index).email();
```

Comment:
Touching the code you could as well extract
`versionAttributeView.readAttributes().history().records().get(index)`  to variable and reuse it below..